aptos-move/framework/aptos-framework/sources/AuctionHouse.move

```diff
@@ -0,0 +1,878 @@
+                auction_init_events: Event::new_event_handle<AuctionInitEvent>(sender),
+                bid_events: Event::new_event_handle<BidEvent>(sender),
+                claim_coins_events: Event::new_event_handle<ClaimCoinsEvent>(sender),
+                claim_token_events: Event::new_event_handle<ClaimTokenEvent>(sender),
+                fps_init_events: Event::new_event_handle<FPSInitEvent>(sender),
+                buy_now_events: Event::new_event_handle<BuyNowEvent>(sender),
+            });
+        };
+
+        if (buy_now_price == 0) {
+            initialize_auction(sender, token_creators, token_collection_names, token_names, min_selling_price, Option::none<u64>(), duration);
+        } else if (buy_now_price == min_selling_price) {
+            initialize_fixed_price_sale(sender, token_creators, token_collection_names, token_names, min_selling_price, duration);
+        } else {
+            initialize_auction(sender, token_creators, token_collection_names, token_names, min_selling_price, Option::some<u64>(buy_now_price), duration);
+        };
+    }
+
+    // initialize fixed price sale
+    public fun initialize_fixed_price_sale(
+        sender: &signer,
+        token_creators: vector<address>,
+        token_collection_names: vector<vector<u8>>,
+        token_names: vector<vector<u8>>,
+        selling_price: u64,
+        duration: u64
+    ) acquires ListingData{
+        let sender_addr = Signer::address_of(sender);
+
+        let start_time = Timestamp::now_microseconds();
+        let listing_id = GUID::id(&GUID::create(sender));
+        let listing_data = borrow_global_mut<ListingData>(sender_addr);
+        let listed_items = &mut listing_data.listed_items;
+
+        Event::emit_event<FPSInitEvent>(
+            &mut listing_data.fps_init_events,
+            FPSInitEvent { id: listing_id, amount: selling_price },
+        );
+
+        Table::add(listed_items, listing_id, ListedItem {
+            min_selling_price: selling_price,
+            buy_now_price: Option::some<u64>(selling_price),
+            duration,
+            start_time,
+            current_bid: Option::none<u64>(),
+            current_bidder: Option::none<address>(),
+            locked_tokens: Option::some(parse_token_ids(sender, token_creators, token_collection_names, token_names)),
+        })
+    }
+
+    // initializes an auction
+    // auction starts with current bidder being the seller and current bid being min_selling_price-1
+    fun initialize_auction(
+        sender: &signer,
+        token_creators: vector<address>,
+        token_collection_names: vector<vector<u8>>,
+        token_names: vector<vector<u8>>,
+        min_selling_price: u64,
+        buy_now_price: Option<u64>,
+        duration: u64
+    ) acquires ListingData {        
+        let sender_addr = Signer::address_of(sender);
+        
+        let start_time = Timestamp::now_microseconds();
+        let listing_id = GUID::id(&GUID::create(sender));
```

Comment:
could we just use the address (Signer::address_of(sender))  as the listing_id?

Reply:
we use listing id to support multiple auctions for the same token at different amounts.